Added header and footer

katy-rex opened the pull request:
@tilikang I changed all the files to php and added the include header and footer.

katy-rex committed:
Added header and footer

Changed all the files to php and added the include header and footer.

tilikang commented on site/Calendar.php:
```diff
@@ -0,0 +1,29 @@
+	</head>
+	<body>
+		<div class="Header">
+			<?php include 'include/header.php';
```
I'd suggest a few changes here:

1- Always put includes at the top of the page unless there's a good reason not to. That way you can easily tell what all the dependencies are in one places instead of needing to hunt around in the code for the includes

2- have your init.php code include header.php, and then you only need to include init.php at the top. As you go forward, there will be more and more libraries you need to include and it becomes a mess if you're doing each one manually. 99% of pages should only have one include, and that's init.php.

3- I'd suggest using `include_once` instead of include. This way if you accidentally include something more than once, PHP is smart enough to ignore the second time.

tilikang commented on site/Calendar.php:
```diff
@@ -0,0 +1,29 @@
+<html>
```
A lot of this code at the top of the page should be in the `PrintHeader` function. The way to decide is to ask yourself if this exact html shows up anywhere else. If it does, it probably needs to go in a function.

This is true of functions in general. If you're writing the exact same code more than once, consider if a function would be better.

In this case, line 7 (the title of the page) isn't 100% exactly the same, but that can still go in the function. I'll leave the exact implementation as an exercise for you, but let me know if you're confused by that.

katy-rex replied:
I updated the header and added both the header and footer to init.php. I figured out the page titles as well. I think I may have messed up with branches though. I made all the edits on Header, which doesn't include the array stuff. If Header gets merged to Master, will those changes trickle down to Array?

tilikang replied:
Sometimes this stuff can get a bit messy and have "conflicts" where github doesn't know what to do. In those cases, you need to manually resolve conflicts which is a pain in the ass, but not a big deal. We can worry about that if it happens later.

tilikang commented on site/Calendar.php:
```diff
@@ -0,0 +1,29 @@
+				PrintHeader();
+			?>
+		</div>
+		<div class="PageTitle">
```
Even this div could go in the `PrintHeader` function if you're using the same format on every page.

katy-rex committed:
Some updates

Figuring out the nuances of branches, and I gotta commit to find something else.

katy-rex committed:
Updated header and page titles

tilikang commented on site/Calendar.php:
```diff
@@ -1,29 +1,7 @@
+		include_once('init.php');
+		$SiteTitle = 'St. Louis Theatre Calendar';
+		$PageTitle = 'Calendar';
+		PrintHeader($SiteTitle, $PageTitle);
```
This isn't a problem, but I just want to note that you can pass strings directly into functions without saving them as a variable first. So you could replace lines 4-6 with:
```
PrintHeader('St. Louis Theatre Calendar', 'Calendar');
```

tilikang commented on site/include/header.php:
```diff
@@ -1,50 +1,69 @@
+	<link rel="stylesheet" type="text/css" href="Style.css">
+	<link href="https://fonts.googleapis.com/css?family=Raleway"
+	rel="stylesheet">
+		<title>'
```
There are some nifty tricks you can use in PHP to make it easier to put variables into strings. what you're doing here is fine, but it's worth learning the tricks, so remind me next time we meet and I can explain some of them.

tilikang commented on site/include/header.php:
```diff
@@ -1,50 +1,69 @@
-					Calendar
-				</p>
+function PrintHeader($SiteTitle, $PageTitle){
+	echo'
```
You don't need to change this, but I'd generally recommend using double quotes for the main echo, and single quotes for things like `rel='stylesheet'` below. The only reason it matters is that the PHP tricks I mention in my next comment only work if you use double quotes.
